lib/gyp/generator/ninja/index.js

```diff
@@ -673,16 +688,21 @@ NinjaMain.prototype.rulesAndTargets = function rulesAndTargets() {
 
   let useCxx = false;
   const ninjas = this.ninjas;
+  const actionNames = {};
+  const targetArch = this.params['target_arch'];
   const ninjaList = this.targetList.map((target, index) => {
+    let targetDict = this.targetDicts[target].configurations[this.config];
     const ninja = new Ninja({
       index: index,
       outDir: this.outDir,
       configDir: this.configDir,
       topDir: this.topDir,
-      target: target,
-      targetDict: this.targetDicts[target].configurations[this.config],
-      ninjas: ninjas,
-      config: this.config
+      target,
```

Comment:
Not that I mind that much, but I still have hopes for transpiling it to ES5 with a minimal AST transformation. Let's keep these changes out for now.